typescript/sdk/src/ism/HyperlaneIsmFactory.ts

```diff
@@ -596,6 +664,63 @@ export async function moduleMatchesConfig(
   return matches;
 }
 
+export async function routingModuleDelta(
+  destination: ChainName,
+  moduleAddress: Address,
+  config: RoutingIsmConfig,
+  multiProvider: MultiProvider,
+  contracts: HyperlaneContracts<ProxyFactoryFactories>,
+  mailbox?: Address,
+): Promise<RoutingIsmDelta> {
+  const signer = multiProvider.getSigner(destination);
+  const provider = multiProvider.getProvider(destination);
+  const routingIsm = DomainRoutingIsm__factory.connect(moduleAddress, provider);
+  const owner = await routingIsm.owner();
+  const deployedDomains = (await routingIsm.domains()).map((domain) =>
+    multiProvider.getChainName(domain.toNumber()),
+  );
+
+  const delta: RoutingIsmDelta = {
+    isOwner: eqAddress(await signer.getAddress(), owner),
+    domainsToUnenroll: [],
+    domainsToEnroll: [],
+  };
+
+  // if owners don't match, we need to transfer ownership
+  if (!eqAddress(owner, normalizeAddress(config.owner)))
+    delta.owner = config.owner;
+  if (config.type === IsmType.FALLBACK_ROUTING) {
+    const client = MailboxClient__factory.connect(moduleAddress, provider);
+    const mailboxAddress = await client.mailbox();
+    if (mailbox && !eqAddress(mailboxAddress, mailbox)) delta.mailbox = mailbox;
+  }
+  // check for exclusion of domains in the config
+  delta.domainsToUnenroll = deployedDomains.filter(
+    (domain) => !Object.keys(config.domains).includes(domain),
+  );
+  // check for inclusion of domains in the config
+  for (const [origin, subConfig] of Object.entries(config.domains)) {
+    if (!deployedDomains.includes(origin)) delta.domainsToEnroll.push(origin);
```

Comment:
should we make these sets?

Reply:
perhaps

Reply:
config.domains are dict.keys and deployedDomains are also dict.keys fetched from onchain, it's implicitly a set